shell/platform/darwin/ios/framework/Headers/FlutterAppDelegate.h

```diff
@@ -24,8 +24,10 @@
  * code as necessary from FlutterAppDelegate.mm.
  */
 FLUTTER_EXPORT
-@interface FlutterAppDelegate
-    : UIResponder <UIApplicationDelegate, FlutterPluginRegistry, FlutterAppLifeCycleProvider>
+@interface FlutterAppDelegate : UIResponder <UIApplicationDelegate,
+                                             FlutterPluginRegistry,
+                                             FlutterAppLifeCycleProvider,
+                                             UNUserNotificationCenterDelegate>
```

Comment:
This would need to be guarded, it's iOS 10+.

I thought @cyanglaz made this stuff happen dynamically?

Reply:
I don't believe this would cause a crash on iOS < 10. The documentation for [Firebase Messaging](https://firebase.google.com/docs/cloud-messaging/ios/client#initialize_firebase_in_your_app) gives instructions on how to add it to your app. Even despite supporting 8+. I also tested this on a 9.0 iOS simulator and the app still compiled and installed on device.

Reply:
Also, @cyanglaz and I talked offline. He was able to dynamically add methods, but not implementing protocols.

Reply:
UNUserNotificationCenterDelegate wasn't added until IOS10.  We'll have to guard this, otherwise warnings will get displayed and people using -Werror will fail.

Reply:
I added an if statement, but I'm confident I didn't implement it correctly. How would I go about doing this?